kiva/celiagg.py

```diff
@@ -868,6 +856,52 @@ def save(self, filename, file_format=None, pil_options=None):
 
         img.save(filename, format=file_format, **pil_options)
 
+    def to_image(self):
+        """ Return the contents of the context as a PIL Image.
+
+        If the graphics context is in BGRA format, it will convert it to
+        RGBA for the image.
+
+        Returns
+        -------
+        img : Image
+            A PIL/Pillow Image object with the data in RGBA format.
+        """
+        try:
+            from PIL import Image
+        except ImportError:
+            raise ImportError("need Pillow to save images")
+
+        pixels = self.gc.array
+        if self.pix_format.startswith('bgra'):
+            # Data is BGRA; Convert to RGBA
+            data = np.empty(pixels.shape, dtype=np.uint8)
+            data[..., 0] = pixels[..., 2]
+            data[..., 1] = pixels[..., 1]
+            data[..., 2] = pixels[..., 0]
+            data[..., 3] = pixels[..., 3]
+        else:
+            data = pixels
+
+        return Image.fromarray(data, 'RGBA')
```

Comment:
It hasn't been a problem yet, but we might want to add support for formats without alpha at some point.